### PR TITLE
go gen improvements

### PR DESCRIPTION
**Pull request by miekg**
Remove the "gen" directory and move directives_generate.go out of it.
Add a build ignore tag so it isn't build by default. Cleanup the go ge
invocations so there are not seen as package docs.

### CODE REVIEW

**Comment by miekg** [comment]
@bwaz I'm still fuzzy on why this 'go gen' needs to be called twice. Why is that?

**Comment by bwaz** [comment]
go generate will provide the path and package name that where it was called from. So if the package layout changes the generated files will change package name and location path as well.  The imports and directives need to parse the same config file to provide output for each use.

**Comment by miekg** [comment]
So the changes I've just pushed, will make `directives_generate.go` break somehow? I'm sorry, but I think I still don't see it. If you can show me some change/code that will break this PR I'm happy to rollback.

File: Makefile [hunk 1]
```diff
@@ -50,7 +50,7 @@ clean:
 
 .PHONY: gen
 gen:
-	go generate ./core/...
+	go generate
```
Comment (bwaz) [inline Makefile]: make sure this actually will generate the z files; delete them first and retry.

Comment (bwaz) [reply Makefile]: I've tried `go generate ./... ` however this caused issues with vendor code - you could for now just explicitly generate the coredns.go

File: directives_generate.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+					} else {
+						mi[items[1]] = middlewarePath + items[2]
+					}
+				}
+
+			}
+		}
+	}
+
+	var orders []int
```
Comment (bwaz) [inline directives_generate.go]: This is unused and redundant what is in the func.... I missed this earlier.

File: Makefile (hunk 1 above)
Comment (miekg) [reply Makefile]: done. Listed `coredns.go` explicitly.

File: directives_generate.go (hunk 2 above)
Comment (miekg) [reply directives_generate.go]: removed.

File: Makefile (hunk 1 above)
Comment (miekg) [reply Makefile]: also removed the z* files and go genned - that works.

**Review by bwaz — APPROVED** [review]

**Pull request merged**